test/runtime/test_apply_requirement_modifications.rb

```diff
@@ -44,7 +44,35 @@ module Runtime
                 execute { Runtime.apply_requirement_modifications(plan) }
 
                 assert_resolution_cancelled do
-                    execute { plan.unmark_permanent_task(requirement_task) }
+                    expect_execution do
```

Comment:
Can you explain what `assert_resolution_cancelled` does? I didn't quite get reading the code